### ISSUE TITLE
Unable to play some media under kernel 5.10.11 when using NFS

### ISSUE THREAD

**Issue by graysky2** [xbmc/xbmc#19147]
### Describe the bug
Upon updating to kernel version 5.10.11, Kodi Matrix (latest git) is unable to playback many videos when they are read from a NFS mount. ~files when they have been scraped and indexed in a shared database.  The same video file is able to playback when it is selected from a non-index video source.~

## Expected Behavior
Any video read from a NFS mount should play.
~All videos indexed in the shared database should playback without error/without crashing Kodi.~

## Actual Behavior
Some videos read from a NFS mount do not play.
~Some videos indexed in the shared database do not playback and instead crash Kodi.~

### To Reproduce
1. Boot into a kernel 5.10.11.
2. Navigate to a video source that is read from a NFS mount.
~2. Navigate to `Movies` within Kodi and select the entry corresponding to one of these files.~
3. Hit the play button.

### Debuglog
* [This](https://gist.github.com/graysky2/88bed8574b666f99c8181dc57e09eb49) debug log is generated from browsing to `Movies>movie title` and the bug is triggered.  **(This is read from a NFS mount).**
* In an attempt to show that nothing is wrong with video file itself, [this](https://gist.github.com/graysky2/2630abaa45f8d2f2966d3b4674a3f2a2) debug log is when I play the _identical file_ from `Videos>Misc>movie.mp4` where, "Misc" is a video source that is not indexed/I have not defined any types for it (ie `Set content>This directory contains` is set to `None`). **(This is not a NFS mount, it is a local mount).**

## Additional context
~This bug is a bit more complex and requires a few lines to explain.  If the target movie is indexed in my shared SQL database, and I try to play it back navigating to `Movies>movie title` under kernel version 5.10.11, the bug is triggered.  If I copy the identical target to the movie, `movie.mp4` to a filepath in Kodi that is not indexed by the database (`Videos>Misc` on my setup), I am able to play it back without error under kernel version 5.10.11.~

If I downgrade to kernel version 5.10.10, I can play the video back regardless of where ~I select it from within Kodi (`Movies>title` or `Videos>Files>Unindexed>movie.mp4`.~ the video is read from, NFS mount or on the local file system.

### Your Environment
Used Operating system:
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: `Arch Linux x86_64`
 - Kodi version: Build from https://github.com/xbmc/xbmc/commit/651468f066dcfec215d7d35967f1a5db8a88f534

**Comment by lrusak** [comment]
Is the MySQL database on the same machine or another machine in the network?

**Comment by graysky2** [comment]
It is on the same machine.

**Comment by fritsch** [comment]
I see that you once play it from a local mount in /mnt - your original log plays it via nfs. Can you make triple sure that you don't just get another IP Address on 5.10 which is not allowed to browse?

And: Booting an older kernel 5.8 or 5.9 or whatever - makes it immediately work?

**Comment by fritsch** [comment]
Can you revert those nfs fixes please: https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.10.11 <- that made it into 5.10.10 - I think you hit a kernel regression.

**Comment by fritsch** [comment]
Also adding libnfs debugging might help. From what I see it just reads garbage via libnfs and as it works with kernel 5.10.10 I think it's the new READ_PLUS support (if kernel is really your only change).

**Comment by graysky2** [comment]
@fritsch - Your observation about NFS mount vs. local mount is a good one!  If I create a new video source that is mapped to my NFS export, I can replicate the bug.  **The database has nothing to do with this**.

I did not see a specific option under `System>Logging>Specify component-specific logging...` for libnfs.  I only see:
```
SMB lib
libcURL lib
FFmpeg lib
JSON-RPC
Audio component
Video component
audo/video timing info
Windowing component
PVR component
EPG component
Announcer component
D-Bus calls
Webserver component
Database component
```

**Comment by graysky2** [comment]
@fritsch - I'm also a little unclear on which to start reverting and then compiling.

What is the best order of these to revert?  You called out three of the six changes .10 to .11 that have to do with NFS specifically:
* There are two mentioning `READ_PLUS`:
https://github.com/torvalds/linux/commit/de82ec8e5e8cba33f84ebef26478b636e94a90fb and https://github.com/torvalds/linux/commit/6533681890902e3b59bbceaea311760b3791c28d
*There is one mentioning `RPC`: https://github.com/torvalds/linux/commit/00ee972739fb2526d3936f1e7ccfc8c91d250c60

Should kernel#1 be reverting the two around `READ_PLUS` at the same time?
Should kernel#2 be just the `RPC` by itself?
If the bug is still there, should I revert all 3 for kernel#3?




To be clear, I see the following commits from the link you posted that have the keyword "NFS" in them:
https://github.com/torvalds/linux/commit/0b6672fd778cd92caed7206ba520a3f056d10484
https://github.com/torvalds/linux/commit/11167454e9cbfa95856fea3f8e5428b4215a534c
https://github.com/torvalds/linux/commit/6ce10b6481cd46040bf3c8f3daec08d3fafa30f4
https://github.com/torvalds/linux/commit/6533681890902e3b59bbceaea311760b3791c28d
https://github.com/torvalds/linux/commit/de82ec8e5e8cba33f84ebef26478b636e94a90fb
https://github.com/torvalds/linux/commit/00ee972739fb2526d3936f1e7ccfc8c91d250c60

Is the best strategy to simply

**Comment by fritsch** [comment]
There is the stable git. You can start on the commit that was 5.10.10 and mark this as good and from there mark the 5.10.11 as bad - here you go.

**Comment by fritsch** [comment]
And from what I see recompile might be needed.

**Comment by graysky2** [comment]
I will clone `git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git` and start the bisect.  Just to crystal clear: is your suggestion:

1. Compile the kernel commit the bisect selects
2. Boot into it
3. Start kodi and test playing the file over NFS

Or

1. Compile the kernel commit the bisect selects
2. Boot into it
3. Recompile libnfs/install new version
4. Start kodi and test playing the file over NFS

**Comment by graysky2** [comment]
@fritsch - The bisect is complete.  As you suspected, https://github.com/torvalds/linux/commit/00ee972739fb2526d3936f1e7ccfc8c91d250c60 is to blame.  What is the most efficient way to report this upstream?  In the past I have used https://bugzilla.kernel.org but responses from the reports logged therein have been hit or miss... mostly ignorned.

**Comment by fritsch** [comment]
@sahlberg has an idea on how to report / handle that best. Especially as this change is not only in 5.10.11 but also in all later versions of the linux kernel.

**Comment by graysky2** [comment]
Sounds good.  Let me know if I can help beyond the bisect.  Thank you very much for your engagement.

Note - Simply reverting that commit and applying it on top of 5.10.12 works.  [Here](https://gist.github.com/graysky2/7a8c72fe38e3031026b06de721b5accc) is the revert patch if others want easy access to it.

**Comment by Maryse47** [comment]
Generally in order to report kernel bug you may look at [MAINTAINERS](https://www.kernel.org/doc/html/latest/process/maintainers.html#maintainers) document and find mailing list for relevant kernel subsystem then send a mail with the details to it. This is a place where kernel developers of specific subsystem are looking.

There is a long read about reporting bugs in kernel [here](https://www.kernel.org/doc/html/latest/admin-guide/reporting-issues.html).

**Comment by graysky2** [comment]
> @sahlberg has an idea on how to report / handle that best. Especially as this change is not only in 5.10.11 but also in all later versions of the linux kernel.

Is there a link to a upstream bug report or ML post you can share?

**Comment by graysky2** [comment]
> @sahlberg has an idea on how to report / handle that best. Especially as this change is not only in 5.10.11 but also in all later versions of the linux kernel.

@fritsch @sahlberg - Is there a link to a upstream bug report or ML post you can share?

**Comment by rks3333** [comment]
I'm seeing the same playback over NFS issue.  I'm running Kodi on an Nvidia Shield and my server is Fedora 33.  Kernel 5.1.10 works fine.  When I play a video with 5.10.11, 5.10.12, and 5.10.13 it starts playing but if I try to fast forward the screen becomes garbled and it never plays the video.

**Comment by graysky2** [comment]
All - The fix was included in the release of the 5.10.15 kernel.  Safe to close from my view.

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.15&id=8e081627f3a7f733a4955ee40b385c972f010f05

**Comment by fritsch** [comment]
For the record:
The Linux NFS client rarely emits an NFS READ **that is not page-
aligned**. All of my testing so far has been with Linux clients, so I
missed this one.
